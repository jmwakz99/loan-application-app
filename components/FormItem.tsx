import { styles } from "@/styles/formItem";
import { FC } from "react";
import { Text, TextInput, View } from "react-native";

type Props = {
  label: string;
  placeholder: string;
  errorMessage?: string;
  onInputChange?: (value: string) => void;
};

const FormItem: FC<Props> = ({
  label,
  placeholder,
  errorMessage = "",
  onInputChange = (_: string) => {},
}) => {
  return (
    <View style={styles.formItemContainer}>
      <Text style={styles.formItemLabel}>{label}</Text>
      <TextInput
        placeholder={placeholder}
        onChangeText={onInputChange}
        style={styles.formItemInput}
      />
      {errorMessage && <Text style={styles.formItemError}>{errorMessage}</Text>}
    </View>
  );
};

export default FormItem;
