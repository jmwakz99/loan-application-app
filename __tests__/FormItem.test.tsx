import { fireEvent, render } from "@testing-library/react-native";

import FormItem from "@/components/FormItem";

it("should render correctly", () => {
  const onInputChangeMock = jest.fn();

  const props = {
    label: "Test label",
    placeholder: "Test placeholder",
    errorMessage: "Test error message",
    onInputChange: onInputChangeMock,
  };

  const formItem = render(<FormItem {...props} />);

  expect(formItem).toMatchSnapshot();
});

it("should call onInputChange when input changes", () => {
  const onInputChangeMock = jest.fn();

  const props = {
    label: "Test label",
    placeholder: "Test placeholder",
    errorMessage: "Test error message",
    onInputChange: onInputChangeMock,
  };

  const { getByPlaceholderText } = render(<FormItem {...props} />);

  fireEvent.changeText(getByPlaceholderText("Test placeholder"), "Test input");

  expect(onInputChangeMock).toHaveBeenCalledWith("Test input");
});
