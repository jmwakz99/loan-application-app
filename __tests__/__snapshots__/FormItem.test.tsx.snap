// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    {
      "gap": 13,
    }
  }
>
  <Text
    style={
      {
        "fontFamily": "Inter-SemiBold",
        "fontSize": 32,
        "fontWeight": "500",
      }
    }
  >
    Test label
  </Text>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Test placeholder"
    style={
      {
        "borderColor": "#D9D9D9",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontFamily": "Inter-Regular",
        "fontSize": 32,
        "height": 92,
        "paddingHorizontal": 26,
        "paddingVertical": 20,
        "width": 594,
      }
    }
  />
  <Text
    style={
      {
        "color": "#ff0000",
        "fontFamily": "Inter-Regular",
        "fontSize": 24,
      }
    }
  >
    Test error message
  </Text>
</View>
`;
